internal/operators/ocs/validations.go

```diff
@@ -99,73 +103,104 @@ func (o *ocsValidator) ValidateOCSRequirements(cluster *models.Cluster) bool {
 		o.log.Info("Setting Operator Status")
 		err := setOperatorStatus(cluster, status)
 		if err != nil {
-			o.log.Fatal("Failed to set Operator status ", err)
+			o.log.Error("Failed to set Operator status ", err)
 			return false
 		}
 		o.log.Info(status)
 		return false
 	} else {
 		for _, host := range hosts { // if the worker nodes >=3 , install OCS on all the worker nodes if they satisfy OCS requirements
 			if host.Role == models.HostRoleWorker {
-				err := o.nodeResources(host, &CPUCount, &TotalRAM, &DiskCount, &HostsWithDisks)
+				err, diskResourcesSufficient := o.nodeResources(cluster, host, &CPUCount, &TotalRAM, &DiskCount, &HostsWithDisks)
 				if err != nil {
-					o.log.Fatal("Error occured while calculating Node requirements ", err)
+					o.log.Error("Error occured while calculating Node requirements ", err)
+					return false
+				} else if !diskResourcesSufficient {
 					return false
 				}
 			}
 		}
 	}
+
+	if DiskCount%3 != 0 {
+		status = "Total disks on the cluster must be a multiple of 3"
+		o.log.Info(status)
+		err := setOperatorStatus(cluster, status)
+		if err != nil {
+			o.log.Error("Failed to set Operator status ", err)
+		}
+		return false
+	}
 	// Hosts with disks must be a multiple of 3
 	o.OCSDisksAvailable = DiskCount
 	canDeployOCS, status := o.validateOCS(o.log, hosts, CPUCount, TotalRAM, DiskCount, HostsWithDisks)
 
+	o.log.Info(status)
 	err := setOperatorStatus(cluster, status)
 	if err != nil {
-		o.log.Fatal("Failed to set Operator status ", err)
+		o.log.Error("Failed to set Operator status ", err)
 		return false
 	}
 
 	return canDeployOCS
 }
 
-func (o *ocsValidator) nodeResources(host *models.Host, CPUCount *int64, TotalRAM *int64, DiskCount *int64, HostsWithDisks *int64) error {
+func (o *ocsValidator) nodeResources(cluster *models.Cluster, host *models.Host, CPUCount *int64, TotalRAM *int64, DiskCount *int64, HostsWithDisks *int64) (error, bool) {
 	var inventory models.Inventory
 	if err := json.Unmarshal([]byte(host.Inventory), &inventory); err != nil {
 		o.log.Errorf("Failed to get inventory from host with id %s", host.ID)
-		return err
+		return err, false
 	}
 
 	disks, err := o.hostApi.GetHostValidDisks(host)
 	if err != nil {
 		o.log.Error("failed to get valid disks due to ", err.Error())
-		return err
+		return err, false
 	}
 
 	if len(disks) > 1 { // OCS must use the non-boot disks
+		diskCPU := int64((len(disks) - 1)) * o.OCSRequiredDiskCPUCount
+		diskRAM := int64((len(disks) - 1)) * o.OCSRequiredDiskRAMGB
+
+		if inventory.CPU.Count < diskCPU || inventory.Memory.UsableBytes < gbToBytes(diskRAM) {
+			status := fmt.Sprint("Resources are not sufficient on host with host ID ", host.ID, " to deploy OCS. A minimum of ", diskCPU, " CPUs, ", diskRAM, "RAMGB is required")
```

Comment:
Please reword to : "Insufficient Resources. Host <id> has <num> disks which requires a minimum of <diskCPU> cpu and <diskRAM>  memory"

or similar. It should include the number of disks on the host + the resources required.